### DISCUSSION TITLE
Inquiry Regarding Pinia, IndexedDB, and $subscribe Compatibility

### DISCUSSION THREAD

**Discussion by FVolral** [vuejs/pinia#2429] [Help and Questions]
Hello,

I am currently developing a mobile application that deals with a significant amount of data, ranging from 50 to 100MB. To simplify my task, I have been using the `pinia-persisted-state-2` plugin, and it seems to be working well. I just need to ensure that the store is ready by using await storeName.$persistedState.isReady().

The major issue I'm encountering is that, over the past few months, the maintainer of this plugin has decided to archive the GitHub repository and redirect users to the use of the `pinia-persisted-state` plugin. Unfortunately, it appears that this plugin may not handle IndexedDB as effectively, as evidenced by two unresolved issues (WON'T FIX) :

Issue #110: [Link to Issue #110](https://github.com/prazdevs/pinia-plugin-persistedstate/issues/110)
Issue #17: [Link to Issue #17](https://github.com/prazdevs/pinia-plugin-persistedstate/issues/17)
So, my question is: Is Pinia truly compatible with IndexedDB, and should I be concerned about potential long-term issues when basing my application on it?

**Comment by Sihamtahi** [comment]
Hello, 
I am facing a similar issue, I am using Pinia Persist for a significant amount of data. Local storage isn't enough to store all my data. I need to use indexedDB, and  have to develop all the CRUD asynchronous operations. Did you find a plugin/lib that facilitate the job ? using indexedDB the smae way as Pinia with Local Storage whithout developping the operations. Thank you in advance.

**Comment by pierresaid** [comment]
You can implement your [own pinia plugin](https://stackoverflow.com/questions/74777210/storing-in-indexeddb-using-pinia/74777649#74777649) to use localforage.

**Reply by Ali-Javanmardi** [reply]
Hi,

It seems localForage has not updated in last two years and there is a discussion about if it is abandoned by its maintainer.
Is there any other option rather than localForage?
